connect/runtime/src/test/java/org/apache/kafka/connect/runtime/ConnectorConfigTest.java

```diff
@@ -434,5 +436,56 @@ public void configure(Map<String, ?> configs) {
         }
     }
 
+    @Test
+    public void testConfigDefOverrideByInitialConfigDef() {
+        String alias = "hdt";
+        String prefix = ConnectorConfig.TRANSFORMS_CONFIG + "." + alias + ".";
+        Map<String, String> props = new HashMap<>();
+        props.put(ConnectorConfig.TRANSFORMS_CONFIG, alias);
+        props.put(prefix + "type", HasDuplicateConfigTransformation.class.getName());
+        ConfigDef def = ConnectorConfig.enrich(MOCK_PLUGINS,
+                new ConfigDef(),
+                props,
+                false);
+        Arrays.asList(PredicatedTransformation.PREDICATE_CONFIG,
+                PredicatedTransformation.NEGATE_CONFIG,
+                HasDuplicateConfigTransformation.MUST_EXIST_KEY)
+                .forEach(key -> {
+                    Assert.assertNotNull(def.configKeys().get(prefix + key));
```

Comment:
I think we should assert that the "predicate" and "negate" config were the implicit ones and not the ones defined by `HasDuplicateConfigTransformation`. I guess we could do this by having `HasDuplicateConfigTransformation`'s configs use different types ("negate is `BOOLEAN` in both) and then asserting the expected types.

Reply:
will roger that